app/models/comment.rb

```diff
@@ -1,4 +1,8 @@
 class Comment < ApplicationRecord
-  belongs_to :user
-  belongs_to :post
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  belongs_to :post, class_name: 'Post', foreign_key: 'post_id'
+
+  def update_post_comments_counter
+    post.update(comments_counter: post.comments.count)
+  end
```

Comment:
-[ ] You could use one of the 👍  [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - *`after_save` to call the `#update_post_counter` method every time a new post has been saved to the database, then make the `update_post_counter` method private. It will allow you to not call that function each time you create a posts 😄 

```ruby
  # app/models/post.rb
  after_save :update_user_posts_counter

  private

  def update_user_posts_counter
    author.update(posts_counter: author.posts.count)
  end
```

Reply:
Thanks for the insight! Using the after_save callback for updating the user's posts counter makes complete sense. I will apply it to the other update methods as well since it make sense to update them each time I create a like or a comment.
Appreciate the suggestion to keep the method private too. Will make these adjustments accordingly.